cmd/tink-worker/container/container.go

```diff
@@ -0,0 +1,111 @@
+package container
+
+import (
+	"context"
+	"path"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+)
+
+type configData struct {
+	name           string
+	pid            string
+	env            []string
+	volumeBindings []string
+	cmd            []string
+	imageName      string
+	registry       string
+	captureLogs    bool
+}
+
+type removeOpts struct {
+	force   bool
+	links   bool
+	volumes bool
+}
+
+// create creates container and returns containerid on success else returns error.
+func create(ctx context.Context, cli client.ContainerAPIClient, data configData) (string, error) {
+	config := &container.Config{
+		Image:        path.Join(data.registry, data.imageName),
+		AttachStdout: true,
+		AttachStderr: true,
+		Cmd:          data.cmd,
+		Tty:          true,
+		Env:          data.env,
+	}
+	if !data.captureLogs {
+		config.AttachStdout = false
+		config.AttachStderr = false
+		config.Tty = false
+	}
+
+	hostConfig := &container.HostConfig{
+		Privileged: true,
+		Binds:      data.volumeBindings,
+	}
+
+	if pidConfig := data.pid; pidConfig != "" {
+		hostConfig.PidMode = container.PidMode(pidConfig)
+	}
+
+	resp, err := cli.ContainerCreate(ctx, config, hostConfig, nil, nil, data.name)
+	if err != nil {
+		return "", errors.Wrap(err, "DOCKER CREATE")
+	}
+	return resp.ID, nil
+}
+
+// start starts container and returns error on failure.
+func start(ctx context.Context, cli client.ContainerAPIClient, id string) error {
+	err := cli.ContainerStart(ctx, id, types.ContainerStartOptions{})
+	if err != nil {
+		return errors.Wrap(err, "DOCKER START")
+	}
+	return nil
+}
+
+// remove delete container and remove volumes, links if flags are set to true. returns error on failure.
+func remove(ctx context.Context, cli client.ContainerAPIClient, id string, opts removeOpts) error {
+	// create options for removing container
+	options := types.ContainerRemoveOptions{
+		Force:         opts.force,
+		RemoveLinks:   opts.links,
+		RemoveVolumes: opts.volumes,
+	}
+
+	// send API call to remove the container
+	err := cli.ContainerRemove(ctx, id, options)
+	if err != nil {
+		return errors.Wrap(err, "DOCKER REMOVE")
+	}
+	return nil
+}
+
+// wait waits on any non "not-running" states and returns container state on success or returns error on failure.
+func wait(ctx context.Context, cli client.ContainerAPIClient, id string) (string, error) {
+	containerState := map[int32]string{
+		0: "SUCCESS",
+		1: "FAILED",
+		2: "TIMEOUT",
+	}
+
+	// Send API call to wait for the container completion
+	wait, errC := cli.ContainerWait(ctx, id, container.WaitConditionNotRunning)
```

Comment:
testing this I found that if I call `func create` then `func wait` I immediately get a `SUCCESS`. What's the expected behavior? should we use `container.WaitConditionNextExit` instead? thoughts?

Reply:
It waits until the container is running that is if it is still in process of creating.

Reply:
Do we want to check the state in our `func wait` before returning? Or have some input parameter a caller can provide that says which states to wait for? Also, with the current structure, there's no way for us to enforce a caller to use the functions in any specific order. We must plan for failure, which one failure case is a user calling create->wait. 
The spec says  "_WaitConditionNotRunning (default) is used to wait for any of the non-running states: "created", "exited", "dead", "removing", or "removed"._"

Reply:
I can add detailed comments when this func needs to be called.